packages/grid/veilid/server/veilid_streamer.py

```diff
@@ -0,0 +1,441 @@
+            del self.buffers[call_id]
+            return response
+        return ResponseType.OK
+
+    async def receive_stream(
+        self,
+        connection: veilid.VeilidAPI,
+        router: veilid.RoutingContext,
+        update: veilid.VeilidUpdate,
+        callback: AsyncReceiveStreamCallback,
+    ) -> None:
+        """Receives a streamed message."""
+        message = update.detail.message
+        prefix = message[:8]
+
+        if prefix == RequestType.STREAM_SINGLE:
+            await self._handle_receive_stream_single(
+                connection, router, update, callback
+            )
+        elif prefix == RequestType.STREAM_START:
+            await self._handle_receive_stream_start(connection, update)
+        elif prefix == RequestType.STREAM_CHUNK:
+            await self._handle_receive_stream_chunk(connection, update)
+        elif prefix == RequestType.STREAM_END:
+            await self._handle_receive_stream_end(connection, router, update, callback)
+        else:
+            logger.error(f"[Bad Message] Message with unknown prefix: {prefix}")
+
+    @retry(veilid.VeilidAPIError, tries=4, delay=1, backoff=2)
+    async def _send_request(
+        self, router: veilid.RoutingContext, dht_key: str, request_data: bytes
+    ) -> None:
+        """Send an app call to the Veilid server and return the response."""
+        async with self._send_request_semaphore:
+            response = await router.app_call(dht_key, request_data)
+            if response != ResponseType.OK:
+                raise Exception("Unexpected response from server")
+
+    async def _send_response(
+        self,
+        connection: veilid.VeilidAPI,
+        call_id: veilid.OperationId,
+        response: bytes,
+    ) -> None:
+        """Send a response to an app call."""
+        async with self._send_response_semaphore:
+            await connection.app_call_reply(call_id, response)
+
+    async def _send_ok_response(
+        self, connection: veilid.VeilidAPI, call_id: veilid.OperationId
+    ) -> None:
+        await self._send_response(connection, call_id, ResponseType.OK)
+
+    async def _send_error_response(
+        self, connection: veilid.VeilidAPI, call_id: veilid.OperationId
+    ) -> None:
+        await self._send_response(connection, call_id, ResponseType.ERROR)
+
+    def _calculate_chunks_count(self, message_size: int) -> int:
+        max_chunk_size = self.chunk_size - self.stream_chunk_header_struct.size
+        total_no_of_chunks = math.ceil(message_size / max_chunk_size)
+        return total_no_of_chunks
+
+    def _get_chunk(
+        self,
+        call_id: bytes,
+        chunk_number: int,
+        message: bytes,
+    ) -> bytes:
+        chunk_header = self.stream_chunk_header_struct.pack(
+            RequestType.STREAM_CHUNK,
+            call_id,
+            chunk_number,
+        )
+        max_actual_message_size = self.chunk_size - self.stream_chunk_header_struct.size
```

Comment:
MAYBE: we could re-name to a common variable name (for max_actual_message_size)
we do the same formula calculation at function `_calculate_chunks_count`
but the variable name is `max_chunk size`

Reply:
Also, we could have the max chunk size calculation as a separate method or present as an instance variable in the class for uniformity?

Reply:
Adding a `self.max_data_in_each_stream_chunk_request` instance variable

Reply:
Keep variable name more descriptive is great.but we could try shortening like 
maybe
1. max_stream_single_msg_size
2. max_stream_chunk_msg_size